### PR TITLE
Parsing import targets from commit message

### PR DESCRIPTION
**Pull request by intrepiditee**
import_target.py is for parsing import targets from commit message. The private constants in import_target.py can be tuned.

### CODE REVIEW

File: import-automation/executor/app/executor/import_target.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    The import with the import name in the only directory touched by the
+    import is executed.
+3) <path to directory containing the manifest>:all
+    All imports in the directory are executed.
+4) all
+    All imports in directories touched by the commit are executed.
+"""
+
+import re
+import typing
```
[the remark this answers is not in the document]
Comment (intrepiditee) [reply import-automation/executor/app/executor/import_target.py:29]: Does an exception mean that it has to be done that way, or can I keep using typing.xxx?

[the remark this answers is not in the document]
Comment (jeffreyoldham) [reply import-automation/executor/app/executor/import_target.py:29]: I am fine with using "typing.List", etc.

File: import-automation/executor/app/executor/import_target.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    _ALLOWED_RELATIVE_IMPORT_NAME_CHARS)
+_IMPORT_NAME_REGEX = r'{}|{}'.format(
+    _ABSOLUTE_IMPORT_NAME_REGEX, _RELATIVE_IMPORT_NAME_REGEX)
+
+
+def get_absolute_import_name(dir_path, import_name):
+    return f'{dir_path}:{import_name}'
+
+
+def absolute_import_name(import_name):
```
[the remark this answers is not in the document]
Comment (intrepiditee) [reply import-automation/executor/app/executor/import_target.py:47]: fixed

File: import-automation/executor/app/executor/import_target.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    return re.fullmatch(_ABSOLUTE_IMPORT_NAME_REGEX, import_name) is not None
+
+
+def relative_import_name(import_name):
```
[the remark this answers is not in the document]
Comment (intrepiditee) [reply import-automation/executor/app/executor/import_target.py:51]: fixed

File: import-automation/executor/app/executor/import_target.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+def split_absolute_import_name(import_name):
+    return import_name.split(':')
+
+
+def get_relative_import_names(import_names):
+    return list(name for name in import_names if relative_import_name(name))
+
+
+def get_absolute_import_names(import_names):
+    return list(name for name in import_names if absolute_import_name(name))
```
[the remark this answers is not in the document]
Comment (intrepiditee) [reply import-automation/executor/app/executor/import_target.py:64]: changed to filter_

File: import-automation/executor/test/test_import_target.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+            import_target.absolute_import_name('scripts/us_fed:treausry/data'))
+        self.assertFalse(
+            import_target.absolute_import_name(':treasury'))
+        self.assertFalse(
+            import_target.absolute_import_name('scripts/us_fed:'))
+        self.assertFalse(
+            import_target.absolute_import_name('scripts/us_fed/treausry'))
+        self.assertFalse(
+            import_target.absolute_import_name(' scripts/us_fed:treausry'))
+        self.assertFalse(
```
[the remark this answers is not in the document]
Comment (intrepiditee) [reply import-automation/executor/test/test_import_target.py:41]: My bad it was misspelled

**Comment by intrepiditee** [comment]
@jeffreyoldham  Thanks. Updated

**Pull request merged**